src/macros.jl

```diff
@@ -875,32 +875,64 @@ Constructs a vector of `LinearConstraint` objects. Similar to `@LinearConstraint
 Constructs a vector of `QuadConstraint` objects. Similar to `@QuadConstraint`, except it accepts multiple constraints as input as long as they are separated by newlines.
 """ :(@QuadConstraints)
 
+"""
+    moi_sense(_error::Function, sense)
+
+Return an expression whose value is an `MOI.OptimizationSense` corresponding
+to `sense`. Sense is either the symbol `:Min` or `:Max`, corresponding
+respectively to `MOI.MinSense` and `MOI.MaxSense` or it is another symbol,
+which should be the name of a variable or expression whose value is `:Min`,
+`:Max` or an `MOI.OptimizationSense`.
+In the last case, the expression throws an error using the `_error`
+function in case the value is a symbol which is not `:Min` nor `:Max`.
+"""
+function moi_sense(_error::Function, sense)
+    if sense == :Min
+        expr = MOI.MinSense
+    elseif sense == :Max
+        expr = MOI.MaxSense
+    else
+        # Refers to a variable that holds the sense.
+        # TODO: Better document this behavior
+        expr = esc(sense)
+    end
+    return :(throw_error_for_symbol_sense($_error, $expr))
+end
+
+# TODO remove for JuMP v0.20
```

Comment:
Actually something like this should stay, because of the `MethodError` principle. What about updating the error message so that it just says what's wrong instead of referring to old JuMP? For example, `"Unexpected sense $value. The sense must be an `OptimizatonSense` or \"Min\" or \"Max\"."` (Also remove the ::Symbol, we want an error if someone gives a sense of `10`.)